tests/Requests/MiddlewareTest.php

```diff
@@ -5,29 +5,37 @@
 namespace OpenEuropa\EPoetry\Tests\Requests;
 
 use GuzzleHttp\Psr7\Response;
+use OpenEuropa\EPoetry\Serializer\RequestsSerializer;
 use OpenEuropa\EPoetry\Tests\Middleware\MockMiddleware;
 use OpenEuropa\EPoetry\Type\CreateRequests;
-use OpenEuropa\EPoetry\Type\LinguisticRequestIn;
-use OpenEuropa\EPoetry\Type\RequestGeneralInfoIn;
 
 /**
  * @internal
  * @coversNothing
  */
-final class MiddlewareTest extends AbstractTest
+final class MiddlewareTest extends AbstractRequestTest
 {
+    /**
+     * Data provider.
+     *
+     * @return mixed
+     */
+    public function responseParsingCases()
+    {
+        return $this->getFixture('request-test.yml');
+    }
+
     /**
      * Test Proxy Ticket in request.
+     *
+     * @dataProvider responseParsingCases
+     *
+     * @param mixed $input
      */
-    public function testResponseParsing()
+    public function testResponseParsing($input)
     {
         // Generate request.
-        $generalInfo = new RequestGeneralInfoIn();
-        $generalInfo->setTitle('Test');
-        $linguisticRequestIn = new LinguisticRequestIn();
-        $linguisticRequestIn->setGeneralInfo($generalInfo);
-        $createRequests = new CreateRequests();
-        $createRequests->setLinguisticRequest($linguisticRequestIn);
+        $createRequests = RequestsSerializer::fromArray($input['request'], CreateRequests::class);
```

Comment:
We shouldn't use the same fixture in two different tests. We did not notice the problem with this because we are not bundling a fixture with both input and expected output.

I'll use the previous library as an example. This [component fixture](https://github.com/ec-europa/oe-poetry-client/blob/master/tests/src/fixtures/arrays/components/details.yml#L1) has both "input" and "output", meaning initial array with which to build a request and the assertions to be ran to test that everything works correctly. 

With should always have both in an entry so that we can run different "test scenarios" on the same tests, that's the ultimate scope of having data providers.

Reply:
Actually in this test we should have a raw XML as input and assert the object values, like we do [here](https://github.com/ec-europa/oe-poetry-client/blob/master/tests/src/fixtures/status.yml#L43) on the old library.

Reply:
I was also a bit concerned about this, but @voidtek and @dxvargas told me that it's fine to use the same feature. Could you please check with them what to do ?

Reply:
We were able to use the same feature because we where not pairing it with assertions, while we should have done that.

Reply:
So, should we fix this in the scope of this ticket ?

Reply:
Fixed.